packages/react-components/react-field/src/components/Field/Field.types.ts

```diff
@@ -54,7 +43,19 @@ export type FieldSlots<T extends FieldControl> = {
 /**
  * Field Props
  */
-export type FieldProps<T extends FieldControl> = ComponentProps<Partial<FieldSlots<T>>, 'control'> & {
+export type FieldProps = Omit<ComponentProps<FieldSlots>, 'children'> & {
+  /**
+   * The Field's child can be a single form control, or a render function that takes the props that should be spread on
+   * a form control.
+   *
+   * All form controls in this library can be used directly as children (such as `<Input>` or `<RadioGroup>`), as well
+   * as intrinsic form controls like `<input>` or `<textarea>`. Custom controls can also be used as long as they
+   * accept FieldChildProps and spread them on the appropriate element.
+   *
+   * For more complex scenarios, a render function can be used to pass the FieldChildProps to the appropriate control.
+   */
+  children?: React.ReactElement<FieldChildProps> | null | ((props: FieldChildProps) => React.ReactNode);
```

Comment:
```suggestion
  children: React.ReactElement<FieldChildProps> | null | ((props: FieldChildProps) => React.ReactNode);
```

Should be `children` required?

Reply:
Making `children` required makes it harder to work with `FieldProps` in cases where you're building out a props object programmatically. While a Field doesn't really make sense without children, it won't technically be broken.